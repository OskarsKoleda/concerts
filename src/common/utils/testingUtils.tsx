import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import type { FC } from "react";
import React from "react";
import RootStore from "../../store/RootStore.ts";

interface ExtendedRenderOptions extends Omit<RenderOptions, "wrapper"> {
  route?: string | null;
  location?: string | null;
  rootStore?: RootStore | null;
}

const withRouter = (children: React.ReactNode, route: string): JSX.Element => {
  return (
    <BrowserRouter future={{ v7_startTransition: true, v7_relativeSplatPath: true }}>
      <Routes>
        <Route path={route} element={children} />
      </Routes>
    </BrowserRouter>
  );
};

export function renderWithProviders(
  UI: React.ReactElement,
  {
    route = "concerts/testing-route",
    location = route,
    rootStore = new RootStore(),
    ...renderOptions
  }: ExtendedRenderOptions = {},
): RenderResult & { rootStore: RootStore | null } {
  if (location) {
    window.history.pushState({}, "", route);
  }

  const WithProvidersWrapper: FC<{ children: React.ReactNode }> = ({ children }) => {
    const UIWithRouter = route ? withRouter(children, route) : <>{children}</>;

    return UIWithRouter;
  };

  return {
    rootStore,
    ...render(UI, { wrapper: WithProvidersWrapper, ...renderOptions }),
  };
}
