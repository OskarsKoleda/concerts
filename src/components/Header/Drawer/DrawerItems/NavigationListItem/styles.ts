import type { SxProps, Theme } from "@mui/material";

import { appTheme } from "../../../../AppProviders/theme.ts";

export const navigationListItemStyles: SxProps<Theme> = {
  "&.Mui-selected": {
    backgroundColor: appTheme.palette.primary.light,

    "&:hover": {
      backgroundColor: appTheme.palette.primary.main,
    },
  },
};
