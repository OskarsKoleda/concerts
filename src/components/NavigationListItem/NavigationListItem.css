a {
  text-decoration: none;
  color: rgb(39, 38, 38);
  background-color: aqua;
}

.active .MuiListItemButton-root {
  background-color: rgb(172, 174, 179);
}

.inactive .MuiListItemButton-root {
  background-color: transparent;
}
