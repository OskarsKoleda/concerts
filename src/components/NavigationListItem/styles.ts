import type { SxProps } from "@mui/material";

export const listItemStyle: SxProps = {
  "&.Mui-selected": {
    backgroundColor: "rgba(0, 0, 0, 0.08)",
    "&:hover": {
      backgroundColor: "rgba(0, 0, 0, 0.12)",
    },
  },
};
