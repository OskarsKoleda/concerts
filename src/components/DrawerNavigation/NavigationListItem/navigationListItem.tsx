import { Link } from "react-router-dom";
import { ListItemButton, ListItemIcon, ListItemText } from "@mui/material";

import React from "react";
import { listItemStyle } from "./styles.ts";

interface ListItemLinkProps {
  selected: boolean;
  label: string;
  to: string;
  icon?: React.ReactElement;
}

export const NavListItem: React.FC<ListItemLinkProps> = ({
  selected,
  label,
  to,
  icon,
}: ListItemLinkProps) => {
  return (
    <ListItemButton component={Link} selected={selected} to={to} sx={listItemStyle}>
      {icon && <ListItemIcon>{icon}</ListItemIcon>}
      <ListItemText primary={label} />
    </ListItemButton>
  );
};
