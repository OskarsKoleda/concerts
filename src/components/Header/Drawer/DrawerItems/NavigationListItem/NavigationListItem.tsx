import { Link } from "react-router-dom";
import { ListItemButton, ListItemIcon, ListItemText } from "@mui/material";

import type { ReactNode } from "react";
import { navigationListItemStyles } from "./styles.ts";

interface NavigationListItemProps {
  selected: boolean;
  label: string;
  to: string;
  icon: ReactNode;
}

const NavigationListItem = ({ selected, label, to, icon }: NavigationListItemProps) => {
  return (
    <ListItemButton component={Link} selected={selected} to={to} sx={navigationListItemStyles}>
      <ListItemIcon>{icon}</ListItemIcon>
      <ListItemText primary={label} />
    </ListItemButton>
  );
};

export default NavigationListItem;
