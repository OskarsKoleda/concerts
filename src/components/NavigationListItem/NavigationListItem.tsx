import { NavLink } from "react-router-dom";
import { ListItem, ListItemText, ListItemButton } from "@mui/material";
import "./NavigationListItem.css";

interface NavListItemProps {
  label: string;
  path: string;
}

const NavListItem: React.FC<NavListItemProps> = ({ label, path }: NavListItemProps) => {
  return (
    <NavLink to={path} className={({ isActive }) => (isActive ? "active" : "inactive")}>
      <ListItemButton>
        <ListItem disablePadding>
          <ListItemText primary={label} />
        </ListItem>
      </ListItemButton>
    </NavLink>
  );
};

export default NavListItem;
