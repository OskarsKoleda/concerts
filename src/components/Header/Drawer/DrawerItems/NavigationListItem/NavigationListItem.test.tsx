import { describe, expect, it } from "vitest";
import { ListItemButton } from "@mui/material";

import { renderWithProviders } from "../../../../../common/utils/testingUtils.tsx";

import NavigationListItem from "./NavigationListItem.tsx";

const props = {
  selected: false,
  label: "Concerts List",
  to: "/events",
  icon: <ListItemButton />,
};

describe("navigationListItem", () => {
  it("should be rendered correctly", () => {
    const { container } = renderWithProviders(<NavigationListItem {...props} />);

    expect(container).toMatchSnapshot();
  });
});
