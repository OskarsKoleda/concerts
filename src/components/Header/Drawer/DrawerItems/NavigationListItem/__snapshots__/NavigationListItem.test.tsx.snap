// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`navigationListItem > should be rendered correctly 1`] = `
<div>
  <a
    class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-u3m3ul-MuiButtonBase-root-MuiListItemButton-root"
    href="/events"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Concerts List
      </span>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </a>
</div>
`;
